File: app/src/main/java/com/stho/software/nyota/utilities/Radian.java
package com.stho.software.nyota.utilities;

import com.stho.software.nyota.utilities.Degree;

/**
 * Created by shoedtke on 20.01.2017.
 */
public class Radian {

    private static final double TWO_PI = Math.PI * 2;


    public static double fromDegrees(double degree) {
        return Math.toRadians(degree);
    }

    public static double toDegrees(double radian) {
        return Math.toDegrees(radian);
    }

    public static double toDegrees180(double radian) {
        return Degree.normalizeTo180(Math.toDegrees(radian));
    }

    public static double fromHour(double hour) {
        return hour * Math.PI / 12;
    }

    public static double normalize(double radian) {
        radian = Math.IEEEremainder(radian, TWO_PI);

        if (radian < 0)
            radian += TWO_PI;

        return radian;
    }
}

File: app/src/main/java/com/stho/nyota/sky/utilities/Radian.kt
package com.stho.nyota.sky.utilities

import kotlin.math.IEEErem

/**
 * Created by shoedtke on 20.01.2017.
 */
object Radian {

    fun fromDegrees(angleInDegree: Double): Double =
        Math.toRadians(angleInDegree)

    fun toDegrees(angleInRadian: Double): Double =
        Math.toDegrees(angleInRadian)

    fun toDegrees180(angleInRadian: Double): Double =
        Degree.normalizeTo180(Math.toDegrees(angleInRadian))

    fun fromHour(hour: Double): Double =
        hour * HOUR_IN_RADIAN

    fun normalize(radian: Double): Double =
        radian.IEEErem(TWO_PI).let {
            when {
                it < 0 -> it + TWO_PI
                else -> it
            }
        }

    fun normalizePi(radian: Double): Double =
        radian.IEEErem(TWO_PI).let {
            when {
                it > PI -> it - TWO_PI
                it < -PI -> it + TWO_PI
                else -> it
            }
        }

    private const val PI = Math.PI
    private const val TWO_PI = 2 * Math.PI
    private const val HOUR_IN_RADIAN = Math.PI / 12
}